### ISSUE TITLE
Typo: Platform Admin Group Policy To Manage Archive Bucket

### ISSUE THREAD

**Issue by TheKoguryo** [oracle-quickstart/oci-landing-zones#75]
When apply templates/enterprise-landing-zone, 'Platform Admin Group Policy To Manage Archive Bucket' for non-prod has a typo.

Generated policy in my tenancy.
Compartment: (root)/OCI-ELZ-CMP-HOME

#### 1. Policy Name: OCI-ELZ-UGP-P-PLT-ADMIN-POLICY-ARCHIVE
Current Policies: 
Allow group OCI-ELZ-UGP-P-PLT-ADMIN to manage buckets in compartment OCI-ELZ-CMP-HOME
Allow group OCI-ELZ-UGP-P-PLT-ADMIN to manage objects in compartment OCI-ELZ-CMP-HOME

After:
Allow group **OCI-ELZ-P-IDT**/OCI-ELZ-UGP-P-PLT-ADMIN to manage buckets in compartment OCI-ELZ-CMP-HOME
Allow group **OCI-ELZ-P-IDT**/OCI-ELZ-UGP-P-PLT-ADMIN to manage objects in compartment OCI-ELZ-CMP-HOME

#### 2. Policy Name: OCI-ELZ-UGP-**N**-PLT-ADMIN-POLICY-ARCHIVE
Current Policies: 
Allow group OCI-ELZ-UGP-**P**-PLT-ADMIN to manage buckets in compartment OCI-ELZ-CMP-HOME
Allow group OCI-ELZ-UGP-**P**-PLT-ADMIN to manage objects in compartment OCI-ELZ-CMP-HOME

After:
Allow group **OCI-ELZ-N-IDT**/OCI-ELZ-UGP-**N**-PLT-ADMIN to manage buckets in compartment OCI-ELZ-CMP-HOME
Allow group **OCI-ELZ-N-IDT**/OCI-ELZ-UGP-**N**-PLT-ADMIN to manage objects in compartment OCI-ELZ-CMP-HOME

I hope that these two policies are fixed.

**Comment by paulmuah** [comment]
Hey @TheKoguryo, many thanks for bringing this to our attention.  I'm tracking the resolution under ISSUE-1951 and I will keep you in the loop.

**Comment by VinayKumar611** [comment]
Hi @TheKoguryo ,

Thanks for your feedback , we are working on the fix. The fix will be implemented  on next release v2.1.3.

**Comment by VinayKumar611** [comment]
Hi @TheKoguryo , 

Issue has been fixed . Please find the [beta release branch ](https://github.com/oracle-quickstart/oci-landing-zones/tree/Typo-Platform-Admin-Group-Policy-%2375) . We will merge the fix in the next release cycle.  We would appreciate it if you could provide us with your feedback on the fix.

Note : Beta Release Branch is not tested throughly.

**Comment by TheKoguryo** [comment]
Hi @VinayKumar611 ,

When I apply that branch, I get the following errors.
```
module.nonprod_environment.module.logging.module.default_log_service_connector[0].oci_sch_service_connector.service_connector: Creation complete after 0s [id=ocid1.serviceconnector.oc1.ap-seoul-1.amaaaaaaw4e6bnyapddppmrs4dmuyuo3q27ga2qwrueeraiyztewxsukdpeq]

Error: 400-InvalidParameter, Compartment {ocid1.compartment.oc1..aaaaaaaa4wa34uimw6ld4ili3htyafasf574rta5hpu3jns7fezkgurehb3q} does not exist or is not part of the policy compartment subtree
Suggestion: Please update the parameter(s) in the Terraform config as per error message Compartment {ocid1.compartment.oc1..aaaaaaaa4wa34uimw6ld4ili3htyafasf574rta5hpu3jns7fezkgurehb3q} does not exist or is not part of the policy compartment subtree
Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/identity_policy
API Reference: https://docs.oracle.com/iaas/api/#/en/identity/20160918/Policy/CreatePolicy
Request Target: POST https://identity.ap-seoul-1.oci.oraclecloud.com/20160918/policies
Provider version: 5.1.0, released on 2023-06-13. This provider is 7 Update(s) behind to current.
Service: Identity Policy
Operation Name: CreatePolicy
OPC request ID: 46b9cecb7280b7e92b020715b867a14b/BB81AF8213117F7DC0CE00F9BE480F52/AD26C99FE94C8689A253BE1D5726A9DF

with module.prod_environment.module.workload.module.workload_osms_dg_policy.oci_identity_policy.policy,
  on ../../modules/policies/main.tf line 9, in resource "oci_identity_policy" "policy" 
   9: resource "oci_identity_policy" "policy" {

    Error: 400-InvalidParameter, Compartment {ocid1.compartment.oc1..aaaaaaaawxblqewj3j35aytt3ia6g7jwzfxpv5fvuw7wnzjatgueeluyaj2a} does not exist or is not part of the policy compartment subtree
Suggestion: Please update the parameter(s) in the Terraform config as per error message Compartment {ocid1.compartment.oc1..aaaaaaaawxblqewj3j35aytt3ia6g7jwzfxpv5fvuw7wnzjatgueeluyaj2a} does not exist or is not part of the policy compartment subtree
Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/identity_policy
API Reference: https://docs.oracle.com/iaas/api/#/en/identity/20160918/Policy/CreatePolicy
Request Target: POST https://identity.ap-seoul-1.oci.oraclecloud.com/20160918/policies
Provider version: 5.1.0, released on 2023-06-13. This provider is 7 Update(s) behind to current.
Service: Identity Policy
Operation Name: CreatePolicy
OPC request ID: 72f75caa15ae0128d99df7cc9390d471/445BBC15D1638E18B8141F6CDA243C53/8796D21B7DFF56EFFD0024B50EE5CB68
  with module.nonprod_environment.module.workload.module.workload_osms_dg_policy.oci_identity_policy.policy,
  on ../../modules/policies/main.tf line 9, in resource "oci_identity_policy" "policy" 
   9: resource "oci_identity_policy" "policy" {
```

You need to change from ${module.workload_compartment.**compartment_id**} to ${module.workload_compartment.**compartment_name**} in file: oci-landing-zones/templates/elz-workload/security.tf
Before
```
    statements = [
      "Allow dynamic-group ${module.workload_osms_dynamic_group.name} to read instance-family in compartment ${module.workload_compartment.compartment_id}",
      "Allow dynamic-group ${module.workload_osms_dynamic_group.name} to use osms-managed-instances in compartment ${module.workload_compartment.compartment_id}"
    ]
```
After
```
    statements = [
      "Allow dynamic-group ${module.workload_osms_dynamic_group.name} to read instance-family in compartment ${module.workload_compartment.compartment_name}",
      "Allow dynamic-group ${module.workload_osms_dynamic_group.name} to use osms-managed-instances in compartment ${module.workload_compartment.compartment_name}"
    ]
```

**Comment by VinayKumar611** [comment]
Hi @TheKoguryo , 

Thanks you for sharing the feedback. We had fixed this Workload OSMS DG policy and updated the beta release branch. Any feedback would be really appreciated . 

 Note : Beta Release Branch is not tested throughly.

**Comment by VinayKumar611** [comment]
Fixed the issue on latest release v2.2.0.
